lib/users/routes.js

```diff
@@ -71,16 +92,126 @@ TODO: inject content into form
 @function
 @name /users/edit/{{id}}
 */
-router.get('/edit/:id', (req, res) => {
-  // this render should populate the form
-  res.render('users/add', {
-    pageTitle: `${config.users.usersHome.title} :: Edit User`,
+router.get('/edit/:id',
+  loginMiddleware({
+    'role': 'superadmin',
+    'redirectTo': '/login',
+    'setReturnTo': '/users/add',
+  }),
+  (req, res, next) => {
+    req.session.editUserId = req.params.id; // eslint-disable-line no-param-reassign
+
+    database('users').where('id', req.params.id)
+    .then(rows => {
+      if (rows.length < 1) {
+        const err = new Error(`User with id ${req.params.id} does not exist`);
+        err.status = 404;
+        res.render('error', {
+          message: err.message,
+          error: err,
+        });
+      }
+
+      users().then(userModel => {
+        // remove passwoord
+        // TODO: password manipulation
+        userModel[0].attributes.splice(1, 1);
+
+        contentTypes.only('users', {
+          'email': {
+            'email': {
+              'value': rows[0].email,
+            },
+          },
+          'role': {
+            'select': {
+              'value': rows[0].role,
+            },
+          },
+        }, userModel)
+        .then(result => {
+          contentTypes.form(result).then(form => {
+            res.render('users/add', {
+              pageTitle: `${config.users.usersHome.title} :: Edit :: ${req.params.id}`,
+              formContents: form.html,
+              validation: form.validation,
+              user: req.user,
+            });
+          });
+        })
+        .catch(error => {
```

Comment:
If you return your promises (which you should be doing and aren't) the errors will bubble and you'll only need one `catch`